gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -113,6 +120,8 @@ public TrinoQueryProperties(
             @JsonProperty("catalogs") Set<String> catalogs,
             @JsonProperty("schemas") Set<String> schemas,
             @JsonProperty("catalogSchemas") Set<String> catalogSchemas,
+            @JsonProperty("procedure") Optional<String> procedure,
+            @JsonProperty("procedureArguments") SerializableCallArgument[] procedureArguments,
```

Comment:
Why do we use array instead of list?

Reply:
This was done because of the use of `requireNonNullElse`, which infers type from its second argument. `[Immutable]List.of()` produces a `List<Object>`, where I needed a `List<SerializableCallArgument>`. It seemed convenient to create an array of the required type inline. 

I'll revert to a list and declare a default of the required type explicitly